tortoise/fields/data.py

```diff
@@ -713,9 +692,51 @@ def to_db_value(
 IntEnumType = TypeVar("IntEnumType", bound=IntEnum)
 
 
+class SmallIntEnumFieldInstance(IntEnumFieldMixin, SmallIntField):
+    pass
+
+
+class IntEnumFieldInstance(IntEnumFieldMixin, IntField):
+    pass
+
+
+class BigIntEnumFieldInstance(IntEnumFieldMixin, BigIntField):
+    pass
+
+
+def get_instance_class(
```

Comment:
It doesn't seem to me that choosing the field type dynamically is a good idea for the following reasons:
- At one moment you can have the value fitting a small int but then you extend your enum and you need to change the type of the column. The migration tools need to count for that, etc.
- This is an implicit behavior and in Python explicit is better than implicit. 
- This introduces new field types that we will need to support! Less things to support is better in general.

Reply:
Ok, Let's keep the lib simple. I will remove the auto choosing function and add an example for how to extend IntEnumField to support int32 instead.